xray/segment.go

```diff
@@ -125,6 +127,17 @@ func BeginSegmentWithSampling(ctx context.Context, name string, r *http.Request,
 			logger.Debugf("SamplingStrategy decided: %t", seg.Sampled)
 			seg.AddRuleName(sd)
 		}
+
```

Comment:
Not sure if moving this piece of code here would be a good idea since `BeginSegmentWithSampling` method is pretty heavy and keeping this code in here would not make this API code more extensible since this piece of code would specifically be useful for http calls. My recommendation would be keep this code inside `htttpTrace` method only or we can create a helper function inside `handler.go` file.